tavla/app/components/NoSSR/ClientOnlyComponent.tsx

```diff
@@ -0,0 +1,29 @@
+import { cloneElement, ReactElement, useEffect, useState } from 'react'
+
+/*
+ *  Some components from the design system are not made for SSR and need to render only on the client.
+ *  This is because they rely on client-side JavaScript to work properly.
+ *
+ *  Wrap the component in this component to ensure it only renders on the client.
+ */
+
+function ClientOnlyComponent({
+    children,
+    ...delegated
+}: {
+    children: ReactElement
+}) {
+    const [isMounted, setIsMounted] = useState(false)
+
+    useEffect(() => {
+        setIsMounted(true)
+    }, [])
+
+    if (!isMounted) {
+        return null
+    }
+
+    return cloneElement(children, delegated)
+}
+
+export default ClientOnlyComponent
```

Comment:
Lurte også på om hele denne komponenten kan skrives om til å bli enda mer generisk som både `TextField` og alt annet kan bruke? Ikke testet, men noe ala dette: 

```
'use client'

function ClientComponentOnly({ children }: { children: React.ReactNode }) {
    return (
        <>
            {children}
        </>
    )
}

export { ClientComponentOnly }
```

Reply:
Hmm, ser kanskje ikke ut som det går: https://github.com/vercel/next.js/discussions/51379

Reply:
Den funker for alt annet😎 Den brukes for textfield også! Eller misforstår jeg hva du mener?

Reply:
Jeg ser man kan bruke `next/dynamic` for å skru av SSR på klientkomponenter også: https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading#skipping-ssr

```
const Tooltip = dynamic(
    () => import('@entur/tooltip').then((mod) => mod.Tooltip),
    { ssr: false },
)
```

Usikker på om dette er noe bedre da, men det er også en mulig løsning. [Den første løsningen Next ](https://nextjs.org/docs/messages/react-hydration-error#solution-1-using-useeffect-to-run-on-the-client-only)nevner er vel den vi har implementer nå?

Reply:
Aha gir mening! Tenkte ikke over den forskjellen, men det gir mening ja. 

Føler vi må undersøke dette litt mer uansett. Vi vet at tooltipen til designsystemet tidligere har vært mye knot med (derfor vi i utgangspunktet hadde laget en egen), men stusser litt over at vi må tvinge TextField til å være klient. Vanskelig å vite om det er oss eller de, men det kan kanskje heller lages en oppgave på, også får man heller inn denne PRen nå? Slipper det å være en blokker 👍 Det er jo beste løsning hvis problemet er at de må være clintside (noe som det ser ut som)

Reply:
Det kan nok være en løsning for å komme oss videre! Jeg kan lage en oppgave som beskriver problemet😎